lib/transmuxer/ts_transmuxer.js

```diff
@@ -243,16 +275,16 @@ shaka.transmuxer.TsTransmuxer = class {
       }
     } catch (e) {
       if (e && e.code == shaka.util.Error.Code.TRANSMUXING_NO_VIDEO_DATA) {
-        return Promise.resolve(new Uint8Array([]));
+        return new Uint8Array([]);
       }
-      return Promise.reject(e);
+      throw e;
```

Comment:
Since this method is not called with `await`, this exception will not be translated into a Promise rejection on `transmux()`, and `transmux()` will throw, too.  Is `transmux()` always called with `await`?

Reply:
I changed the entire implementation to reduce duplicate code.